nwb_conversion_tools/json_schema_utils.py

```diff
@@ -64,22 +67,29 @@ def get_schema_from_method_signature(class_method, exclude=None):
         list="array"
     )
 
-    for param in inspect.signature(class_method).parameters.values():
-        if param.name not in exclude + ['self']:
+    for param_name, param in inspect.signature(class_method).parameters.items():
+        if param_name not in exclude:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                param_type = [
+                    annotation_json_type_map[arg_type.__name__] for arg_type in param.annotation.__args__
+                    if arg_type.__name__ in annotation_json_type_map
+                ]
```

Comment:
```suggestion
                param_type = [
                    annotation_json_type_map[arg_type] for arg_type in param.annotation.__args__
                    if arg_type in annotation_json_type_map
                ]
```

does this work?

Reply:
Unfortunately, no. The actual `arg_type` object in this case is a `type` class, and does not map properly onto the keys of the `annotation_json_type_map`. The `__name__ `of the `arg_type` does, however.

Reply:
ah, right, but we can fix that

Reply:
by changing the dict to

```python
    annotation_json_type_map = {
        bool: "boolean",
        str: "string",
        int: "number",
        float: "number",
        dict: "object",
        list: "array"
    }
```

Reply:
Ahh, clever... Adding now